src/timing.h

```diff
@@ -351,6 +353,9 @@ STATIC_INLINE void jl_timing_block_start(jl_timing_block_t *block) {
 
     uint64_t t = cycleclock(); (void)t;
     _COUNTS_START(&block->counts_ctx, t);
+#ifdef USE_TIMING_COUNTS
+    block->counts_ctx.t0 = t;
```

Comment:
Why not put this in `_COUNTS_START`?

Reply:
Ah of course, you don't want this to run in `_jl_timing_block_destroy`

Reply:
I would be open to this re-factoring:
- `_COUNTS_START   => _COUNTS_START (only called once) + _COUNTS_RESUME (resumes self timer)`
- `_COUNTS_STOP    => _COUNTS_PAUSE (stops self timer)`
- `_COUNTS_DESTROY => _COUNTS_END (stops timer and adds counts to totals)`

Hopefully that also makes it clearer that _COUNTS_END should be called conditionally if the block was running, not unconditionally to destroy the object itself